application/LicenseAppService.go

```diff
@@ -2,24 +2,59 @@ package application
 
 import (
 	"authz/domain/contracts"
+	"authz/domain/model"
+	"authz/domain/services"
 	"context"
 )
 
 // LicenseAppService the handler for seat related endpoints.
 type LicenseAppService struct {
-	accessRepo *contracts.AccessRepository
-	ctx        context.Context
+	accessRepo    contracts.AccessRepository
+	seatRepo      contracts.SeatLicenseRepository
+	principalRepo contracts.PrincipalRepository
+	ctx           context.Context
+}
+
+type ModifySeatAssignmentRequest struct {
+	Requestor model.Principal
+	OrgId     string
+	ServiceId string
+	Assign    []string
+	Unassign  []string
 }
 
 // NewLicenseAppService ctor.
-func (s *LicenseAppService) NewLicenseAppService(accessRepo *contracts.AccessRepository) *LicenseAppService {
+func NewLicenseAppService(accessRepo contracts.AccessRepository, seatRepo contracts.SeatLicenseRepository, principalRepo contracts.PrincipalRepository) *LicenseAppService {
 	return &LicenseAppService{
-		accessRepo: accessRepo,
-		ctx:        context.Background(),
+		accessRepo:    accessRepo,
+		seatRepo:      seatRepo,
+		principalRepo: principalRepo,
+		ctx:           context.Background(),
 	}
 }
 
 // ModifySeats TODO
-func (s *LicenseAppService) ModifySeats(_ string) error {
-	return nil
+func (s *LicenseAppService) ModifySeats(req ModifySeatAssignmentRequest) error {
+	evt := model.ModifySeatAssignmentEvent{
+		Request: model.Request{
+			Requestor: req.Requestor,
+		},
+		Org:     model.Organization{Id: req.OrgId},
+		Service: model.Service{Id: req.ServiceId},
+	}
+
+	var err error
+	evt.Assign, err = s.principalRepo.GetByIDs(req.Assign)
+	if err != nil {
+		return err
+	}
+
+	evt.UnAssign, err = s.principalRepo.GetByIDs(req.Unassign)
+	if err != nil {
+		return err
+	}
+
+	seatService := services.NewSeatLicenseService(s.seatRepo, s.accessRepo)
```

Comment:
In line with previous comments, it feels like these repos should be defined in the bootstrap/main somewhere and "injected" into the SeatLicenceService, rather than being passed down from here. (Assuming that all ModifySeatAssignmentRequests will be dealt with by the same repos.)

Reply:
I think this is mainly a lifetime concern and definitely agree: the current setup is kind of leaking that the repos are singletons, and that doesn't make sense long-term. I assume we'll be bringing in a DI framework at some point, that can be set up in bootstrap and manage lifetimes of individual repos that may be singletons, per request, etc.